ndcube/ndcube.py

```diff
@@ -607,11 +609,105 @@ def _fill_in_crop_nones(self, lower_corner, upper_corner, wcs, crop_by_values):
 
         return lower_corner, upper_corner
 
-    def _bounding_box_to_points(self, lower_corner_values, upper_corner_values, wcs):
+    def _bounding_box_to_points(self, lower_corner_values, upper_corner_values,
+                                axis_correlation_matrix):
         """
         Convert two corners of a bounding box to the points of all corners.
         """
-        return tuple(itertools.product(*zip(lower_corner_values, upper_corner_values)))
+        # Calculate which world axes share multiple pixel axes.
+        world_n_dim, pixel_n_dim = axis_correlation_matrix.shape
+        world_axes = np.arange(world_n_dim)
+        world_n_dep_dim = axis_correlation_matrix.sum(axis=1)
+        # If all world coordinates are 1D, bounding box is simple.
```

Comment:
The bounding box is also simple if n_world == n_pixel

Reply:
Good point.  I'll add a check for this and call the original implementation if `True`